src/System.CodeDom/src/System/CodeDom/Compiler/Executor.cs

```diff
@@ -91,6 +91,8 @@ public static int ExecWaitWithCapture(IntPtr userToken, string cmd, string curre
                         throw new ExternalException(SR.Format(SR.ExecTimeout, cmd), WAIT_TIMEOUT);
                     }
 
+                    p.WaitForExit();
```

Comment:
why is this needed? we already wait for process to exit line above

Reply:
See the docs for WaitForExit(int).

Reply:
per docs, it returns true when process did exit and then we wait again, when it is false we throw `ExternalException`. Am I missing something?

Reply:
Ah I see.. "To ensure that asynchronous event handling has been completed, call the WaitForExit() overload that takes no parameter after receiving a true from this overload"